src/components/Layout/Layout.tsx

```diff
@@ -42,6 +46,19 @@ export const Layout = forwardRef(function Layout(
   const metaUrl = url ? url : basePath + router.asPath;
   if (!router.isReady) return <></>;
 
+  // [platform] will always be the very first subpath right?
+  // when using `router.asPath` it returns a string that starts with a '/'
+  // To get the "platform" the client was trying to visit, we have to get the string at index 1
+  // Doing this because when visiting a 404 page, there is no `router.query.platform`, so we have
+  // to check where the user was trying to visit from
+  const asPathPlatform = router.asPath.split('/')[1] as Platform;
```

Comment:
As far as platform logic goes, with the new IA we are expecting that this platform is passed into every page except maybe the main home page (which defaults to JS so maybe we just want to pass in JS and keep it consistent).  If this is the case do we want to make this parameter required and do away with these potential edge cases?  @hbuchel do you know under what circumstances the platform wouldn't be passed into the Layout component?  Do we still have platform agnostic pages?

Reply:
I think at this point only the pages under /gen2 are platform agnostic. For the "normal" docs, there were too many cases of nested pages that needed platform variations so they moved them all under /[platform]

Reply:
@jacoblogan @hbuchel, does that mean we want to have the `Layout` component take in the platform? Sorry confused on what the action item is for this discussion.

Jacob, what do you mean by making the parameter required, should it be required for a specific component?

Reply:
sorry what I meant by required was that in the Layout component the platform parameter is optional.  What I'd like to avoid is for us to be figuring out the platform in multiple locations but I might be overreacting.  With the gen2 pages being platform agnostic we will probably need this logic here.